src/main/java/monique/parser/Parser.java

```diff
@@ -0,0 +1,169 @@
+package monique.parser;
+
+import java.util.Arrays;
+import java.util.Set;
+
+import monique.command.*;
+import monique.exception.IllegalDateFormatException;
+import monique.exception.ParseException;
+import monique.task.Deadline;
+import monique.task.Event;
+import monique.task.Task;
+import monique.task.ToDo;
+
+/**
+ * The <code>Parser</code> class processes user input and returns the corresponding <code>Command</code> object.
+ * It handles various commands and task types, parsing the input to create appropriate command objects.
+ */
+public class Parser {
+    private static final Set<String> commands = Set.of("list", "mark", "unmark", "bye", "/commands", "delete", "find");
+    private static final Set<String> taskTypes = Set.of("todo", "deadline", "event");
+
+    /**
+     * Parses the given command string and returns the corresponding <code>Command</code> object.
+     * The method identifies the command type and creates the appropriate command object with the provided parameters.
+     *
+     * @param fullCommand The full command string to be parsed.
+     * @return The <code>Command</code> object corresponding to the parsed command.
+     */
+    public static Command parse(String fullCommand) {
+        String firstWord = fullCommand.split(" ")[0];
+        boolean hasSecondWord = fullCommand.split(" ").length > 1;
+        Command command = null;
+
+        if (commands.contains(firstWord)) {
+            switch (firstWord) {
+            case "bye": {
+                command = new ByeCommand();
+                break;
+            }
+            case "list": {
+                command = new ListCommand();
+                break;
+            }
+            case "mark": {
+                //minus one bc 0-based indexing
+                try {
+                    if (!hasSecondWord) {
+                        throw new ParseException();
+                    }
+                    int taskNum = Integer.parseInt(fullCommand.split("mark ")[1]) -1;
+                    command = new MarkCommand(taskNum);
+                    break;
```

Comment:
I think this break is unnecessary because of the finally break?

Reply:
Thanks! I've fixed it :)